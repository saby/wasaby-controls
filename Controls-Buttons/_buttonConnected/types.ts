import type { ObjectMeta } from 'Types/meta';

type TCommandOptions = Record<string, unknown>;

/**
 * Интерфейс объекта, описывающего действия.
 * @public
 */
export interface IActionConfig<RuntimeInterface extends object> {
    /**
     * Уникальный идентификатор действия
     * @remark это значение пишется в настройки действия во фрейме
     */
    id: string;

    /**
     * Человекопонятное название действия
     */
    title: string;
    /**
     * Категория действия
     */
    category?: string;
    /**
     * Иконка действия
     */
    icon?: string;
    /**
     * Алиас до модуля-исполнителя команды
     * @interface
     */
    commandName?: string;
    /**
     * Опции, которые пережаются в commandName первым аргументом
     */
    commandOptions?: TCommandOptions;
    /**
     * Мета-тип атрибутов команды
     */
    propsMeta?: ObjectMeta<RuntimeInterface>;
}

/**
 * Интерфейс модуля-исполнителя команды
 */
export interface IActionExecutor {
    /**
     * Метод, запускающий выполнение действия
     * @param commandOptions опции действия, заданные в его описании
     * @param editorProps опции действия, настроенные в конструкторе через {@link Controls-editors/propertyGrid:PropertyGrid PropertyGrid}
     */
    execute(commandOptions: TCommandOptions, editorProps: Record<string, unknown>);
}
