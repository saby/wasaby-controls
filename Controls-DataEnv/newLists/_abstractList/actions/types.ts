import type * as marker from './types/marker';
import type * as operationsPanel from './types/operationsPanel';
import type * as root from './types/root';
import type * as search from './types/search';
import type * as selection from './types/selection';
import type * as source from './types/source';
import type * as expandCollapse from './types/expandCollapse';
import type * as filter from './types/filter';

import type * as interactorCore from './types/interactorCore';

export type {
    marker,
    operationsPanel,
    root,
    search,
    selection,
    source,
    expandCollapse,
    filter,
    interactorCore,
};

/**
 * Тип действия, доступного в любом списке, независимо от типа ViewModel, к которой он подключен (web/mobile).
 */
export type TAnyAbstractAction =
    | source.TAnySourceAction
    | expandCollapse.TAnyExpandCollapseAction
    | marker.TAnyMarkerAction
    | filter.TAnyFilterAction
    | selection.TAnySelectionAction
    | operationsPanel.TAnyOperationsPanelAction
    | root.TAnyRootAction
    | search.TAnySearchAction
    | interactorCore.TAnyInteractorCoreAction;
