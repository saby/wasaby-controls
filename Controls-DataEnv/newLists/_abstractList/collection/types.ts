export { TCollectionType } from './types/TCollectionType';
