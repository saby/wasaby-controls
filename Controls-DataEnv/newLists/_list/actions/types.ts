import type { TAbstractListActions } from 'Controls-DataEnv/abstractList';

import type * as marker from './types/marker';
import type * as operationsPanel from './types/operationsPanel';
import type * as root from './types/root';
import type * as search from './types/search';
import type * as selection from './types/selection';
import type * as items from './types/items';
import type * as filter from './types/filter';
import type * as source from './types/source';
import type * as expandCollapse from './types/expandCollapse';
import type * as complexUpdate from './types/complexUpdate';

export type {
    marker,
    operationsPanel,
    root,
    expandCollapse,
    search,
    selection,
    items,
    filter,
    source,
    complexUpdate,
};

/**
 * Тип действия, доступного в WEB списке.
 */
export type TAnyListAction =
    | TAbstractListActions.TAnyAbstractAction
    | expandCollapse.TAnyExpandCollapseAction
    | marker.TAnyMarkerAction
    | selection.TAnySelectionAction
    | operationsPanel.TAnyOperationsPanelAction
    | root.TAnyRootAction
    | search.TAnySearchAction
    | items.TAnyItemsAction
    | source.TAnySourceAction
    | filter.TAnyFilterAction
    | complexUpdate.TAnyComplexUpdateAction;
