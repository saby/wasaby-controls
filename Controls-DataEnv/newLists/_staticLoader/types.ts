import type { IAbstractListState } from 'Controls-DataEnv/abstractList';

export type TState = Pick<IAbstractListState, 'viewMode' | 'isLatestInteractorVersion'>;

export interface TDependencyDescriptor<T extends keyof TState> {
    prop: T;
    value: TState[T][] | ((state: TState) => boolean);
}

export type TUI_Dependencies = Record<string, TDependencyDescriptor<keyof TState>[]>;
