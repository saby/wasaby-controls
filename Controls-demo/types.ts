import { IHeaderCell } from 'Controls/grid';
import { RecordSet } from 'Types/collection';

export interface IHeader extends IHeaderCell {
    title: string;
}

export type TItemsReadyCallback = (items: RecordSet) => void;
export type TRoot = string | number | null;

export interface INavigation {
    source: string;
    view: string;
    sourceConfig: {
        pageSize: number;
        page: number;
        hasMore?: boolean;
    };
    viewConfig?: {
        pagingMode?: string;
        maxCountValue?: number;
    };
}
