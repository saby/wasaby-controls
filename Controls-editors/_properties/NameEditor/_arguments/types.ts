import { ObjectMeta } from 'Meta/types';

const PARAMETERS_FIELD = 'parameters';
const RETURN_VALUE_FIELD = 'return_value';

interface IFunctionCallMeta {
    arguments: ObjectMeta | null;
    returnValue: ObjectMeta | null;
}

export { IFunctionCallMeta, PARAMETERS_FIELD, RETURN_VALUE_FIELD };
