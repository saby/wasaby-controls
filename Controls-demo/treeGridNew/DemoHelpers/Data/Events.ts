import { IColumn } from 'Controls/grid';
import { IData } from '../Interface';

export const Events = {
    getColumns: (): IColumn[] => {
        return [
            {
                displayProperty: 'title',
                width: '',
            },
        ];
    },
    getDataBeforeExpanded: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataBeforeCollapsed: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataCollapsedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataExpandedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
};
