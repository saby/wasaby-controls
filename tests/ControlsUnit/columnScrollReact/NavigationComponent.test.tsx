/**
 * @jest-environment jsdom
 */
jest.mock('Controls/_columnScrollReact/Navigation/ArrowButtons', () => {
    return {
        __esModule: true,
        default: (props) => {
            return (
                <div data-qa="mocked_ArrowButtonsNavigationComponent">
                    <div data-qa="mocked_arrows_props">{JSON.stringify(props, null, 4)}</div>
                    <div
                        data-qa="mocked_arrow_left"
                        onClick={() => {
                            return props.onArrowClick?.('backward');
                        }}
                    />
                    <div
                        data-qa="mocked_arrow_right"
                        onClick={() => {
                            return props.onArrowClick?.('forward');
                        }}
                    />
                </div>
            );
        },
    };
});

jest.mock('Controls/_columnScrollReact/Navigation/Scrollbar', () => {
    return {
        __esModule: true,
        default: (props) => {
            return (
                <div data-qa="mocked_ScrollbarNavigationComponent">
                    {JSON.stringify(props, undefined, 4)}
                </div>
            );
        },
    };
});

import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';

import * as React from 'react';
import { fireEvent, waitFor } from '@testing-library/react';

import NavigationComponent, {
    INavigationComponentProps,
} from 'Controls/_columnScrollReact/NavigationComponent';
import {
    ColumnScrollContext,
    IColumnScrollContext,
    PrivateContextUserSymbol,
} from 'Controls/_columnScrollReact/context/ColumnScrollContext';
import { SELECTORS } from './SelectorsForTests';
import { renderAnyComponent } from './forTestsOnly/helpers';
import { AnimationSteps } from './forTestsOnly/animationSteps';
import { EdgeState } from 'Controls/_columnScrollReact/common/types';

function FakeContextProviderForNavigation(props: {
    children: JSX.Element;
    position?: number;
    leftEdgeState?: IColumnScrollContext['leftEdgeState'];
    rightEdgeState?: IColumnScrollContext['rightEdgeState'];
    setPositionCallback?: IColumnScrollContext['setPosition'];
}): JSX.Element {
    const contextRefForHandlersOnly = React.useRef<IColumnScrollContext>();

    const context = React.useMemo<IColumnScrollContext>(
        () => ({
            contextRefForHandlersOnly,
            SELECTORS,
            viewPortWidth: 600,
            contentWidth: 1600,
            startFixedWidth: 300,
            endFixedWidth: 0,
            position: props.position || 20.124,
            leftEdgeState: props.leftEdgeState,
            rightEdgeState: props.rightEdgeState,
            setPosition: props.setPositionCallback || jest.fn(),
            isNeedByWidth: true,

            isScrollbarDragging: false,
            updateSizes: jest.fn(),
            isMobile: false,
            isMobileScrolling: false,
            setIsMobileScrolling: jest.fn(),
            mobileSmoothedScrollPosition: undefined,
            setIsScrollbarDragging: jest.fn(),
        }),
        []
    );

    contextRefForHandlersOnly.current = context;

    return (
        <ColumnScrollContext.Provider value={context}>
            {props.children}
        </ColumnScrollContext.Provider>
    );
}

describe('ControlsUnit/columnScrollReact/NavigationComponent', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        container.remove();
        container = null;
    });

    describe('with context and consumer.', () => {
        describe('snapshots with different prop[mode].', () => {
            const testCases: INavigationComponentProps['mode'][] = [
                'arrows',
                'scrollbar',
                undefined,
            ];

            testCases.forEach((mode: INavigationComponentProps['mode']) => {
                it(`mode=${mode}`, () => {
                    const { asFragment } = renderAnyComponent(
                        <FakeContextProviderForNavigation>
                            <NavigationComponent mode={mode} />
                        </FakeContextProviderForNavigation>,
                        { container }
                    );
                    expect(asFragment()).toMatchSnapshot();
                });
            });
        });

        describe('should pass right props isLeftEnabled and isRightEnabled, depends on context props leftEdgeState and rightEdgeState.', () => {
            const testCases: [EdgeState, EdgeState][] = [
                [EdgeState.Visible, EdgeState.Invisible],
                [EdgeState.Invisible, EdgeState.Invisible],
                [EdgeState.Invisible, EdgeState.Visible],
                ...(AnimationSteps.map((step) => {
                    return [step.leftEdgeState, step.rightEdgeState];
                }) as [EdgeState, EdgeState][]),
            ];

            testCases.forEach(([leftEdgeState, rightEdgeState]) => {
                it(`leftEdgeState=${leftEdgeState}, rightEdgeState=${rightEdgeState}`, () => {
                    const { asFragment } = renderAnyComponent(
                        <FakeContextProviderForNavigation
                            leftEdgeState={leftEdgeState}
                            rightEdgeState={rightEdgeState}
                        >
                            <NavigationComponent mode="arrows" />
                        </FakeContextProviderForNavigation>,
                        { container }
                    );
                    expect(asFragment()).toMatchSnapshot();
                });
            });
        });

        it('should set new position t context on arrow click', async () => {
            const setPositionCallback = jest.fn();

            const { getByQA } = renderAnyComponent(
                <FakeContextProviderForNavigation
                    position={253}
                    setPositionCallback={setPositionCallback}
                >
                    <NavigationComponent mode="arrows" />
                </FakeContextProviderForNavigation>,
                { container }
            );

            const rightButton = getByQA('mocked_arrow_right');

            fireEvent.click(rightButton);
            await waitFor(() => {
                expect(setPositionCallback).toBeCalledTimes(1);
                expect(setPositionCallback).lastCalledWith(523, true, PrivateContextUserSymbol);
            });

            const leftButton = getByQA('mocked_arrow_left');

            fireEvent.click(leftButton);
            await waitFor(() => {
                expect(setPositionCallback).toBeCalledTimes(2);
                expect(setPositionCallback).lastCalledWith(0, true, PrivateContextUserSymbol);
            });
        });
    });
});
