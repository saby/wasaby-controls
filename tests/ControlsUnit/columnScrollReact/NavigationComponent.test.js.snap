// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. should pass right props isLeftEnabled and isRightEnabled, depends on context props leftEdgeState and rightEdgeState. leftEdgeState=AnimatedFromInvisibleToVisible, rightEdgeState=AnimatedInvisible 1`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ArrowButtonsNavigationComponent"
  >
    <div
      data-qa="mocked_arrows_props"
    >
      {
    "isLeftEnabled": false,
    "isRightEnabled": true,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123"
}
    </div>
    <div
      data-qa="mocked_arrow_left"
    />
    <div
      data-qa="mocked_arrow_right"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. should pass right props isLeftEnabled and isRightEnabled, depends on context props leftEdgeState and rightEdgeState. leftEdgeState=AnimatedFromVisibleToInvisible, rightEdgeState=AnimatedInvisible 1`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ArrowButtonsNavigationComponent"
  >
    <div
      data-qa="mocked_arrows_props"
    >
      {
    "isLeftEnabled": false,
    "isRightEnabled": true,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123"
}
    </div>
    <div
      data-qa="mocked_arrow_left"
    />
    <div
      data-qa="mocked_arrow_right"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. should pass right props isLeftEnabled and isRightEnabled, depends on context props leftEdgeState and rightEdgeState. leftEdgeState=AnimatedInvisible, rightEdgeState=AnimatedFromInvisibleToVisible 1`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ArrowButtonsNavigationComponent"
  >
    <div
      data-qa="mocked_arrows_props"
    >
      {
    "isLeftEnabled": true,
    "isRightEnabled": false,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123"
}
    </div>
    <div
      data-qa="mocked_arrow_left"
    />
    <div
      data-qa="mocked_arrow_right"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. should pass right props isLeftEnabled and isRightEnabled, depends on context props leftEdgeState and rightEdgeState. leftEdgeState=AnimatedInvisible, rightEdgeState=AnimatedFromVisibleToInvisible 1`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ArrowButtonsNavigationComponent"
  >
    <div
      data-qa="mocked_arrows_props"
    >
      {
    "isLeftEnabled": true,
    "isRightEnabled": false,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123"
}
    </div>
    <div
      data-qa="mocked_arrow_left"
    />
    <div
      data-qa="mocked_arrow_right"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. should pass right props isLeftEnabled and isRightEnabled, depends on context props leftEdgeState and rightEdgeState. leftEdgeState=AnimatedInvisible, rightEdgeState=AnimatedInvisible 1`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ArrowButtonsNavigationComponent"
  >
    <div
      data-qa="mocked_arrows_props"
    >
      {
    "isLeftEnabled": true,
    "isRightEnabled": true,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123"
}
    </div>
    <div
      data-qa="mocked_arrow_left"
    />
    <div
      data-qa="mocked_arrow_right"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. should pass right props isLeftEnabled and isRightEnabled, depends on context props leftEdgeState and rightEdgeState. leftEdgeState=AnimatedInvisible, rightEdgeState=AnimatedInvisible 2`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ArrowButtonsNavigationComponent"
  >
    <div
      data-qa="mocked_arrows_props"
    >
      {
    "isLeftEnabled": true,
    "isRightEnabled": true,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123"
}
    </div>
    <div
      data-qa="mocked_arrow_left"
    />
    <div
      data-qa="mocked_arrow_right"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. should pass right props isLeftEnabled and isRightEnabled, depends on context props leftEdgeState and rightEdgeState. leftEdgeState=Invisible, rightEdgeState=Invisible 1`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ArrowButtonsNavigationComponent"
  >
    <div
      data-qa="mocked_arrows_props"
    >
      {
    "isLeftEnabled": true,
    "isRightEnabled": true,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123"
}
    </div>
    <div
      data-qa="mocked_arrow_left"
    />
    <div
      data-qa="mocked_arrow_right"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. should pass right props isLeftEnabled and isRightEnabled, depends on context props leftEdgeState and rightEdgeState. leftEdgeState=Invisible, rightEdgeState=Visible 1`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ArrowButtonsNavigationComponent"
  >
    <div
      data-qa="mocked_arrows_props"
    >
      {
    "isLeftEnabled": true,
    "isRightEnabled": false,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123"
}
    </div>
    <div
      data-qa="mocked_arrow_left"
    />
    <div
      data-qa="mocked_arrow_right"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. should pass right props isLeftEnabled and isRightEnabled, depends on context props leftEdgeState and rightEdgeState. leftEdgeState=Visible, rightEdgeState=Invisible 1`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ArrowButtonsNavigationComponent"
  >
    <div
      data-qa="mocked_arrows_props"
    >
      {
    "isLeftEnabled": false,
    "isRightEnabled": true,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123"
}
    </div>
    <div
      data-qa="mocked_arrow_left"
    />
    <div
      data-qa="mocked_arrow_right"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. snapshots with different prop[mode]. mode=arrows 1`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ArrowButtonsNavigationComponent"
  >
    <div
      data-qa="mocked_arrows_props"
    >
      {
    "isLeftEnabled": false,
    "isRightEnabled": false,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123"
}
    </div>
    <div
      data-qa="mocked_arrow_left"
    />
    <div
      data-qa="mocked_arrow_right"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. snapshots with different prop[mode]. mode=scrollbar 1`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ScrollbarNavigationComponent"
  >
    {
    "readOnly": false,
    "position": 20.124,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123",
    "viewPortWidth": 600,
    "contentWidth": 1600,
    "startFixedWidth": 300,
    "endFixedWidth": 0
}
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/columnScrollReact/NavigationComponent with context and consumer. snapshots with different prop[mode]. mode=undefined 1`] = `
<DocumentFragment>
  <div
    data-qa="mocked_ScrollbarNavigationComponent"
  >
    {
    "readOnly": false,
    "position": 20.124,
    "className": "js-controls-ColumnScrollReact_notDragScrollable_guid123",
    "viewPortWidth": 600,
    "contentWidth": 1600,
    "startFixedWidth": 300,
    "endFixedWidth": 0
}
  </div>
</DocumentFragment>
`;
